### PR TITLE
`--ignore` feature implemented.

### PR DESCRIPTION
**Pull request by noel-yap**

### CODE REVIEW

**Comment by nilslice** [comment]
Closes #3. Thank you @noel-yap! I've merged this into #30 and will be cutting a new release shortly.

File: parse.go [hunk 1]
```diff
@@ -293,29 +285,72 @@ func getUpdatedLock() (*Protolock, error) {
...
 			return nil
 		}
+
 		// skip to next if is a directory
 		if info.IsDir() {
 			return nil
 		}
 
+		// if ignore, skip
+		if ignores != "" {
```
Comment (sargun) [inline parse.go]: This should have probably been broken out into its own function

File: parse.go [hunk 2]
```diff
@@ -293,29 +285,72 @@ func getUpdatedLock() (*Protolock, error) {
...
+			for _, ignore := range strings.Split(ignores, ",") {
+				rootIgnore := root + "/" + ignore
```
Comment (sargun) [inline parse.go]: filepath.Join should be used, not this.

File: parse.go [hunk 3]
```diff
@@ -293,29 +285,72 @@ func getUpdatedLock() (*Protolock, error) {
...
+				stat, err := os.Stat(rootIgnore)
+				if err == nil {
+					if stat.IsDir() && !strings.HasSuffix(rootIgnore, "/") {
```
Comment (sargun) [inline parse.go]: filepath.abs / filepath.rel comes in handy here.

File: parse.go [hunk 4]
```diff
@@ -293,29 +285,72 @@ func getUpdatedLock() (*Protolock, error) {
...
+						rootIgnore += "/"
+					}
+
+					if strings.HasPrefix(path, rootIgnore) {
```
Comment (sargun) [inline parse.go]: https://golang.org/pkg/path/filepath/#HasPrefix

File: parse_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package protolock
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (sargun) [inline parse_test.go]: You really should use a vendoring solution if you use 3rd party code.

**Comment by nilslice** [comment]
@sargun take a look at the more recent PR #30 i believe. Some of this was fixed but please feel free to make edits off of master.

File: parse_test.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+func TestGetProtoFilesFiltersDirectories(t *testing.T) {
+	files, err := getProtoFiles("testdata/getProtoFiles", "")
+	assert.NoError(t, err)
```
Comment (sargun) [inline parse_test.go]: This should probably be a "require.NoError"

File: parse.go (hunk 4 above)
Comment (noel-yap) [reply parse.go]: HasPrefix exists for historical compatibility and should not be used.

Deprecated: HasPrefix does not respect path boundaries and does not ignore case when required.

**Comment by nilslice** [comment]
@sargun - we're glad to have you looking through the code. All feedback and suggestions are welcomed :) in the future, please make sure you're making comments on the latest code though, to take special note to avoid commenting on closed PRs or issues, as your suggestions may have already been considered or implemented. 

Additionally, code speaks much louder than comments! If you think things should be changed, please submit a PR and discuss on that thread which changes you've made. 

Thank you!

**Comment by noel-yap** [comment]
@nilslice , I gave the link to the closed PR to @sargun since I requested a code review from him. Any non-conformance to how things are done in this project stem from my own actions, not Sargun's. I should've been more explicit in this PR regarding my request which stemmed from the fact that I'm brand new to Go and I knew Sargun, someone I work with, would be more nit-picky than you would be since I'm guessing you would lean towards politeness. I'm hoping no offense has been taken by you through my actions or with this response and that you know we're all here are being motivated by good intentions.

**Comment by nilslice** [comment]
@noel-yap, I appreciate the explanation. It's difficult sometimes to tell how outside contributors will respond to code reviews and as a maintainer I typically try to stand up for the project and the people who contribute code. 

I also want to make sure @sargun knows I appreciate the review, and didn't mean to show any disrespect! Open source is tough, as we all sit behind our screens, some with less context as others :)

> I'm hoping no offense has been taken by you through my actions or with this response and that you know we're all here are being motivated by good intentions.

No offense taken at all, and I hope the same for you both. Good intentions all around :)

Glad to have you and your co-workers helping with this project & want to have the highest quality code as possible -- please keep the reviews, PRs, ideas, comments coming 👍

**Comment by noel-yap** [comment]
@nilslice , I completely understand and appreciate where you're coming from regarding different people's responses to code reviews. In fact, as soon as I started seeing Sargun's comments, I wondered how you would take them considering you were missing the context of my request to him.

I'm glad this is all cleared up :-)

**Comment by nilslice** [comment]
> I'm glad this is all cleared up :-)

Likewise. Have a great weekend!

**Pull request merged**